### PR TITLE
Fix x-ghaf-hw-test agent label

### PR DESCRIPTION
**Pull request by vjuntunen**
Use agent labal matching target device.

Enable riscv HW tests to ghaf-main-pipeline

### CODE REVIEW

**Review by azbeleva — APPROVED** [review]

File: tests/x-ghaf-hw-test.groovy [hunk 1]
```diff
@@ -111,7 +112,7 @@ def ghaf_robot_test(String testname='boot') {
 ////////////////////////////////////////////////////////////////////////////////
 
 pipeline {
-  agent { label "${params.getOrDefault('LABEL', DEF_LABEL)}" }
+  agent { label "${params.LABEL}" }
```
Comment (henrirosten) [inline tests/x-ghaf-hw-test.groovy:115]: I would prefer using:

```
label "${params.getOrDefault('LABEL', DEF_LABEL)}"
```

here instead. The reason is, params.LABEL is undefined the first time you trigger this pipeline after jenkins service reboot.

File: tests/x-ghaf-hw-test.groovy [hunk 2]
```diff
@@ -15,6 +15,7 @@ def CONF_FILE_PATH = '/etc/jenkins/test_config.json'
 properties([
   parameters([
     string(name: 'IMG_URL', defaultValue: 'https://ghaf-jenkins-controller-dev.northeurope.cloudapp.azure.com/artifacts/ghaf-release-pipeline/build_8-commit_5c270677069b96cc43ae2578a72ece272d7e1a37/packages.aarch64-linux.nvidia-jetson-orin-nx-debug/sd-image/nixos-sd-image-24.11.20240802.c488d21-aarch64-linux.img.zst', description: 'Target image url'),
+    string(name: 'LABEL', defaultValue: 'orin-nx', description: "Target testagent need to match with target image device! 'orin-nx', 'orin-agx', 'nuc', 'riscv' or 'lenovo-x1'"),
```
Comment (henrirosten) [inline tests/x-ghaf-hw-test.groovy:18]: I'm pretty sure there would be a way to derive the LABEL value from the IMG_URL, removing the need to have the LABEL parameter altogether.

There's some risk that whoever triggers this manually, will type an incorrect value to the LABEL argument, which could have unwanted consequences.

File: tests/x-ghaf-hw-test.groovy (hunk 1 above)
Comment (henrirosten) [reply tests/x-ghaf-hw-test.groovy:115]: Also on line 9, you now have:

```
def DEF_LABEL = 'testagent'
```

if you use this variable (DEF_LABEL), please provide a valid default value or perhaps a null or empty string.

**Review by henrirosten — CHANGES_REQUESTED** [review]

File: tests/x-ghaf-hw-test.groovy (hunk 1 above)
Comment (vjuntunen) [reply tests/x-ghaf-hw-test.groovy:115]: Maybe empty is better so you need to actual set right agent when starting this job.

File: tests/x-ghaf-hw-test.groovy (hunk 2 above)
Comment (vjuntunen) [reply tests/x-ghaf-hw-test.groovy:18]: Yes I was thinking that also but it is just a quick fix for Mariia so she can run stuff. Should be improved when we have that real dev env.

**Review by henrirosten — APPROVED** [review]

**Pull request merged**